elki/src/main/java/de/lmu/ifi/dbs/elki/algorithm/clustering/kmeans/initialization/PredefinedInitialMeans.java

```diff
@@ -94,9 +94,7 @@ public void setInitialClusters(List<? extends Cluster<? extends MeanModel>> init
    */
   public void setInitialMeans(double[][] initialMeans) {
     double[][] vecs = new double[initialMeans.length][];
-    for(int i = 0; i < initialMeans.length; ++i) {
-      vecs[i] = initialMeans[i]; // TODO: clone?
-    }
+    System.arraycopy(initialMeans, 0, vecs, 0, initialMeans.length);
```

Comment:
I noticed the TODO, but I am not sure I understand what you still would want to accomplish with it. Or do your prefer a clone over an array copy?